sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_document_model_administration_client.py

```diff
@@ -623,18 +623,30 @@ def callback(raw_response, _, headers):  # pylint: disable=unused-argument
 
         if self._api_version == DocumentAnalysisApiVersion.V2022_08_31:
             raise ValueError("Method 'begin_build_document_classifier()' is only available for API version "
-                             "V2023_02_28_PREVIEW and later")
+                             "V2023_07_31 and later")
         cls = kwargs.pop("cls", callback)
         continuation_token = kwargs.pop("continuation_token", None)
         polling_interval = kwargs.pop("polling_interval", self._client._config.polling_interval)
         if classifier_id is None:
             classifier_id = str(uuid.uuid4())
 
+        _doc_types = {}
+        for doc, details in doc_types.items():
+            _source = None
+            if hasattr(details.source, "prefix"):
+                _source = self._generated_models.ClassifierDocumentTypeDetails(
+                    azure_blob_source=self._generated_models.AzureBlobContentSource(
+                        container_url=details.source.container_url, prefix=details.source.prefix))
+            elif hasattr(details.source, "file_list"):
+                _source = self._generated_models.ClassifierDocumentTypeDetails(
+                    azure_blob_file_list_source=self._generated_models.AzureBlobFileListContentSource(
+                        container_url=details.source.container_url, file_list=details.source.file_list))
+            _doc_types[doc] = _source
```

Comment:
This will be a big improvement! Thanks for pointing it out. Tracking issue to do in a separate PR: https://github.com/Azure/azure-sdk-for-python/issues/31276